### ISSUE TITLE
Lammps input control

### ISSUE THREAD

**Issue by LVC14C15** [pyiron/pyiron_atomistics#598]
**Summary**

Hello, 

I would like to report some issues regarding the input control for lammps. 

One of my routine to minimize structure is the following:
minimization with fire
then
box/relax + cg 
then again fire

Currently, to my understanding, there is two ways to do it in pyiron:
1st: run three different jobs for each minimization, but then you have to call the last structure. 

2nd: modify the input script via the input control command and add all the features (fix, compute...). This option is for me the "best" (avoiding to have a lot of jobs). 

However, for both, I encounter several issues. You can't apparently decide the order of the command you modified (fix, compute...) that means that you can have:
```
min_style fire directly following by min_style cg 
```
without the minimize 0.0 1.0e-8 10000 20000 command that should follow

Lammps will read the two command "simultaneously" and this will lead to an error 

As an example if I do this:  
```
job.input.control['min_style'] = 'fire'
job.input.control['minimize'] = '0.0 1.0e-8 10000 20000'
job.input.control['fix___3'] = 'all box/relax y 0.0 vmax 0.001'
job.input.control['min_style___cg'] = ''   
job.input.control['minimize'] = '0.0 1.0e-8 10000 20000'
```

My input looks like this: 
```
fix 3 all box/relax y 0.0 vmax 0.001
min_style fire
min_style cg 
minimize 0.0 1.0e-8 10000 20000
```
which is not what I want and will not works also.

An other issues is link to parameters: 
you can notice that I did this: `job.input.contro**l['fix___3']** = 'all box/relax y 0.0 vmax 0.001'`
The problem is if I use a the command `job.input.control**['fix']** = '**3** all box/relax y 0.0 vmax 0.001'` it will erase my previous fix commands. 

I hope I was clear :) 
Tell me if not and also if I missed something here!
All the best,
Dimitri

**Comment by jan-janssen** [comment]
Ok, there are multiple answers to this. 

**Abstract Answer**: For pyiron to be able to follow the simulation workflow we want every `run` or `minimize` command to be a separate job. So you could write a restart file and then create a new job from the existing one. 

**Expert Answer**: pyiron is written in a way that you can always manipulate the input that is communicated to the simulation code: 
```
job.input.control.load_string(input_str="""\
units metal
dimension 3
boundary p p p
atom_style atomic
read_data structure.inp
include potential.inp
fix ensemble all box/relax iso 0.0
variable dumptime equal 100
variable thermotime equal 100
min_style cg
dump 1 all custom ${dumptime} dump.out id type xsu ysu zsu fx fy fz vx vy vz
dump_modify 1 sort id format line "%d %d %20.15g %20.15g %20.15g %20.15g %20.15g %20.15g %20.15g %20.15g %20.15g"
thermo_style custom step temp pe etotal pxx pxy pxz pyy pyz pzz vol
thermo_modify format float %20.15g
thermo ${thermotime}
minimize 0.0 0.0001 100000 10000000
min_style cg
minimize 0.0 1.0e-8 10000 20000
"""                          
)
```

**Hacky Answer**: To get as close as possible to the regular pyiron syntax, you can use: 
```
job.calc_minimize(pressure=0.0)
job.input.control["min_style"] = "fire"
job.input.control._insert(line_number=16, data_dict={"Parameter": ['min_style'], "Value": ['cg'], "Comment": [""]})
job.input.control._insert(line_number=17, data_dict={"Parameter": ['minimize'], "Value": ['0.0 1.0e-8 10000 20000'], "Comment": [""]})
```

I hope this helps you to continue your calculation, still I guess it is a general topic we should consider during the development.

**Comment by LVC14C15** [comment]
Hi Jan, 

Thanks for your answer, it helps!
The problem to my point of view is to set up all the fix/compute command. The command  job.calc_minimize(pressure=0.0) is really nice for simple calculation, but when one needs additional commands such as fix and/or compute,  the expert answer you describe is the best option I think.
Thanks again for your help ;)

Best Regards, 
Dimitri
